modules/apps/depot/depot-web/src/main/resources/META-INF/resources/js/ConnectedSiteDropdownPropsTransformer.js

```diff
@@ -25,7 +23,25 @@ class ConnectedSiteDropdownDefaultEventHandler extends DefaultEventHandler {
 		) {
 			submitForm(document.hrefFm, itemData.disconnectSiteActionURL);
 		}
-	}
-}
+	},
+};
+
+export default function propsTransformer({items, ...otherProps}) {
+	return {
+		...otherProps,
+		items: items.map((item) => {
+			return {
+				...item,
+				onClick(event) {
```

Comment:
@wincent, any idea on how we could provide this boilerplate so devs won't have to code this over and over? 

The whole mapping to add the `onClick` handler, and then figure out which action to run depending on the `data.action` (or other) data attribute.

This is more or less what the `defaultEventHandler` pattern was doing. So, without exactly doing that, I'm afraid we will need to do something to avoid this repetition over and over... 🤔

Reply:
We discussed this on Slack and (without having had time to look in depth at what `DefaultEventHandler` did) it seems ok to tolerate the duplication for now in order to unblock this.

One idea that occurs to me is that if we start using an analog for `DefaultEventHandler` we might be able to move away from the [class structure](https://github.com/liferay/liferay-portal/blob/0dabb62a733eb28bcfc0752fd20e9ac8300ba472/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/DefaultEventHandler.es.js) which involves an inheritance hierarchy and a number of methods towards something more granular. In this concrete instance, it doesn't seem to be doing much at all.

Reply:
OKay, tabling this for the future